db: TestTableStats nil pointer dereference

Seen on CI: https://travis-ci.org/github/cockroachdb/pebble/jobs/700210545

```
--- FAIL: TestTableStats (0.02s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd06a33]
goroutine 17382 [running]:
testing.tRunner.func1(0xc003ac5600)
	/home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:874 +0x69f
panic(0xde5dc0, 0x16e3fc0)
	/home/travis/.gimme/versions/go1.13.12.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/pebble/internal/datadriven.runTestInternal.func1.1(0xc00e520388)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/internal/datadriven/datadriven.go:101 +0x196
panic(0xde5dc0, 0x16e3fc0)
	/home/travis/.gimme/versions/go1.13.12.linux.amd64/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/pebble.TestTableStats.func3(0xc00e520388, 0xc00e520380, 0xc00e520388)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/table_stats_test.go:95 +0x223
github.com/cockroachdb/pebble/internal/datadriven.runTestInternal.func1(0xc00e520388, 0xc016295ef8, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/internal/datadriven/datadriven.go:104 +0x8a
github.com/cockroachdb/pebble/internal/datadriven.runTestInternal(0xc003ac5600, 0xea5adf, 0x14, 0x1015360, 0xc0126e5318, 0xc016295ef8, 0x0)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/internal/datadriven/datadriven.go:109 +0x103
github.com/cockroachdb/pebble/internal/datadriven.RunTest(0xc003ac5600, 0xea5adf, 0x14, 0xc0000bbef8)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/internal/datadriven/datadriven.go:79 +0x1a7
github.com/cockroachdb/pebble.TestTableStats(0xc003ac5600)
	/home/travis/gopath/src/github.com/cockroachdb/pebble/table_stats_test.go:43 +0x546
testing.tRunner(0xc003ac5600, 0xf0d8d0)
	/home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:909 +0x19a
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.13.12.linux.amd64/src/testing/testing.go:960 +0x652
```

--- a/table_stats_test.go
+++ b/table_stats_test.go
@@ -6,7 +6,6 @@ package pebble
 
 import (
 	"fmt"
-	"sync"
 	"testing"
 
 	"github.com/cockroachdb/pebble/internal/base"
@@ -18,17 +17,13 @@ import (
 
 func TestTableStats(t *testing.T) {
 	fs := vfs.NewMem()
-	var loadedInfo struct {
-		sync.Mutex
-		*TableStatsInfo
-	}
+	// loadedInfo is protected by d.mu.
+	var loadedInfo *TableStatsInfo
 	opts := &Options{
 		FS: fs,
 		EventListener: EventListener{
 			TableStatsLoaded: func(info TableStatsInfo) {
-				loadedInfo.Lock()
-				loadedInfo.TableStatsInfo = &info
-				loadedInfo.Unlock()
+				loadedInfo = &info
 			},
 		},
 	}
@@ -57,12 +52,11 @@ func TestTableStats(t *testing.T) {
 
 		case "reopen":
 			require.NoError(t, d.Close())
+			loadedInfo = nil
+
 			// Open using existing file system.
 			d, err = Open("", opts)
 			require.NoError(t, err)
-			loadedInfo.Lock()
-			loadedInfo.TableStatsInfo = nil
-			loadedInfo.Unlock()
 			return ""
 
 		case "batch":
@@ -91,8 +85,9 @@ func TestTableStats(t *testing.T) {
 			for d.mu.tableStats.loading || !d.mu.tableStats.loadedInitial {
 				d.mu.tableStats.cond.Wait()
 			}
+			s := loadedInfo.String()
 			d.mu.Unlock()
-			return loadedInfo.String()
+			return s
 
 		case "compact":
 			if err := runCompactCmd(td, d); err != nil {

